src/components/text-field.js

```diff
@@ -38,89 +56,228 @@ const _textFieldClasses = ({ container, inputSize, success, error }) => ({
+    'border-b border-error-500': focus && error,
+    'pointer-events-none bg-secondary-100 text-secondary-400 border-secondary-400': disabled,
+  }),
+  input: clsx(`${classes.input} outline-none mb-0.5 bg-transparent font-normal tracking-md text-xs ${styles.borderlessInput}`, {
+    'text-secondary-900': isEdited,
+    'text-secondary-600': !isEdited,
+  }),
 })
 
-const TextField  = ({ classes, size, inputProps, label, maxLength, helperText, success, error, required, disabled, deleteButton, onChange, onClick, onDelete, onSubmit, ...rest }) => {
+const renderLabel = ({ label, required, textFieldClasses }) => (
+  <div className='flex flex-row'>
+    {label && <p className={textFieldClasses.label}>{label}</p>}
+    {required && <span className='flex flex-row ml-5px text-error-500'>*</span>}
+  </div>
+)
+const renderFooter = ({ helperText, maxLength, value, textFieldClasses }) => (
+  <div className={`flex flex-row ${helperText ? 'justify-between' : 'justify-end'}`}>
+    {helperText && <p className={textFieldClasses.helperText}>{helperText}</p>}
+    {maxLength && <p className={textFieldClasses.wordCount}>{value.length || 0}/{maxLength}</p>}
+  </div>
+)
+
+const TextField  = ({
+  classes, variant, size, inputProps, label, maxLength, helperText, success, error,
+  required, disabled, deleteButton, onChange, onClick, onDelete, onSubmit, linkedFields,
+  ...rest
+}) => {
   const [filled, setFilled] = useState(false)
   const [value, setValue] = useState(false)
   const [focus, setFocus] = useState(false)
+  const [isEdited, setIsEdited] = useState(false)
+  const [linkedVals, setLinkedVals] = useState([])
+  const [linkedIncompleteError, setLinkedIncompleteError] = useState(false)
+
+  const inputID = counter()
   const { root, input, container } = classes
-  const inputSize = _inputSize({ size })
-  const textFieldClasses = _textFieldClasses({ container, inputSize, success, error })
-  const inputBaseClasses = _inputBaseClasses({ label, inputSize, focus, success, error, root, input, filled, disabled })
+  const inputSize = _inputSize({ size, variant })
+  const textFieldClasses = _textFieldClasses({ container, inputSize, success, error: error || linkedIncompleteError, linkedFields })
+  const generateVariants = ({ linked }) => ({
+    default: _inputBaseClasses({ label, inputSize, focus, success, error: error || linkedIncompleteError, root, input, filled, disabled, linked }),
+    borderless: _borderlessClasses({ classes, isEdited, error, focus, disabled }),
+  })
+
+  const handleLinkedChange = (e, i, inputID) => {
+    if (e.target.value.length >= 1) {
+      const next = document.getElementById(`linked-${inputID}-${i+2}`)
+      if (next) {
+        next.focus()
+        setFocus(i+2)
+      }
+    }
+
+    if ((i + 1) === linkedFields && linkedVals[i]) {
+      return onChange(linkedVals.join(''))
+    }
 
-  const handleChange = (val) => {
-    setValue(val)
+    const v = [...linkedVals]
+    v.splice(i, 1, e.target.value)
```

Comment:
Would be nice to have some functionalities to filter out special characters

Reply:
think would leave this up to the users